### ISSUE TITLE
XDG support

### ISSUE THREAD

**Issue by fenuks** [TASEmulators/fceux#412]
fceux currently uses ~/.fceux to store its data. Adding support for XDG was already mentioned in an old [TODO file](https://github.com/TASEmulators/fceux/blob/master/attic/TODO-SDL-2.1.6.md#xdg-standardization-of-config), therefore I won't explain what XDG is for. Said file already sketches general strategy how this could be achieved without breaking existing installations.

**Comment by keithbowes** [comment]
It would be especially great as switching between <= 2.2.x and >= 2.3.x is difficult due to the differences in the config.  Really, this should have been done before 2.3.0 (either using different directories or different file names, as AFAICT, there's no way to specify the location of the config file on the command line in any version).

**Comment by mjbudd77** [comment]
The Qt version checks for a $FCEUX_HOME environment variable to be defined, which will be used instead of $HOME when choosing where to use config files. I know it's not what you looking for exactly here. But you may find it useful.

**Comment by fenuks** [comment]
Thanks, this is useful information. Of course full XDG conformance is much more than that, but it allows at least to move fceux directory to other place. I'll add this workaround to [Arch Wiki](https://wiki.archlinux.org/title/XDG_Base_Directory).

If there is no opposition to adhering to XDG Base Specification, I could try adding it for Qt version, since it has [QStandardPaths](https://doc.qt.io/qt-5/qstandardpaths.html) greatly simplifying the task.

**Comment by mjbudd77** [comment]
I don't think you will be able to follow the XDG spec exactly. The application config code is not setup to be spread out. Also, most windows users would not want this as they prefer the base fceu directory being with the executable path. I have considered adding a dialog window that lets the user choose how the user wants the base directory selected (similar to how mesen does it).

**Comment by fenuks** [comment]
GUI dialog is good idea. Other than that, XDG compliance could be Unix specific, if that would make WIndows users complain. Strictly speaking, the specification allows storing fceux base directory alongside executable if you tweak environment variables, e.g. XDG_CONFIG_HOME="$(pwd)" XDG_DATA_HOME="$(pwd)" ./fceux. This effectively achieves portable mode without extra code.
